### PR TITLE
Comm manual fixes

### PR DESCRIPTION
**Pull request by VincentXiao-jpg**
Added the DS_Store ignore back to frontend

### CODE REVIEW

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+        "processed": "Only available to a Bachelor of Commerce as part of a single || double-degree"
+    """
+    return "COMM#"
+
+def COMM_1040():
+    """
+        "original": "Prerequisite: Students must be in Good Academic Standing<br/><br/>",
+        "processed": "must be in Good Academic Standing"
+    """
+    return ""
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:74]: Can you return this as a warning? Refer to your condition below as an example

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    """
+        "original": "Pre-requisite: COMM1100 and excludes MGMT1101. Only available to single and double degree Business School students in Term 2. It will be offered to non-Business School students in Term 3.<br/><br/>",
+        "processed": "COMM1100 &&"
+    """
+
+    COURSES["COMM1150"]["exclusions"]["MGMT1101"] = 1
+    return {
+        "original": conditions["original"],
+        "processed": "COMM1100",
+    }
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:134]: Can you add warning about the Term 2 and Term 3 thing?

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+        "processed": "Only available to single && double degree Business School in Term 1. It will be offered to non-Business School in Terms 2 && 3"
+    """
+
+    return {
+        "original": conditions["original"],
+        "processed": "",
+        "warning": "Students enrolled in 3764 (Eng/Comm), Actuarial Studies or Economics programs (in both single and double degree mode) are not permitted to enrol."
+    }
+
+# def COMM_1120(conditions):
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:100]: Best to comment the return out as well (idk how the behaviour is working here)

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+        "processed": "Only available to single && double degree Business School in Term 1. It will be offered to non-Business School in Terms 2 && 3"
+    """
+
+    return {
+        "original": conditions["original"],
+        "processed": conditions["processed"],
+        "warning": "Only available to single and double degree Business School in Term 1. It will be offered to non-Business School in Terms 2 and 3."
+    }
+
+# def COMM_1140(conditions):
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:112]: Likewise here

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+        "processed": "COMM1100 &&"
+    """
+
+    COURSES["COMM1150"]["exclusions"]["MGMT1101"] = 1
+    return {
+        "original": conditions["original"],
+        "processed": "COMM1100",
+    }
+
+# def COMM_1170(conditions):
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:136]: Likewise here and below, remember to comment out the entire function or else there's a bunch of floating returns (which probably breaks the code when we run it)

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+def COMM_2222(conditions):
+    """
+        "original": "Pre-requisite: In Business degree, COMM6000 Career Accelerator or COMM1999 First Year Portfolio, WAM 65+, minimum 72 UOC completed, room in degree for course, Good academic standing.<br/>Excluded: Co-op students<br/>It is recommended a progression check is completed prior to enrolling.<br/><br/>",
+        "processed": "In Business degree || COMM6000 Career Accelerator || COMM1999 First Year Portfolio, 65WAM, 72UOC , room in degree for course, Good academic standing"
+    """
+
+    return  {
+        "original": conditions["original"],
+        "processed": "(COMM6000 || COMM1999) && ZBUS && 65WAM && 72UOC"
+    }
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:202]: Change ZBUS to ZBUS#

Add in warning field about good standing and how co-op students are excluded

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+
+def COMM_2233():
+    """
+        "original": "Completed 48 UOC, Good Standing, completed COMM6000 or COMM1999, and have space in degree for course.<br/>Its recommended to seek a progression check prior to application. Visit Career Accelerator page on Business School website for more info.<br/><br/>",
+        "processed": "48UOC || Good Standing || COMM6000 || COMM1999 && have space in degree for course. Its recommended to seek a progression check prior to application. Visit Career Accelerator page on Business School website for more info"
+    """
+
+    return "(COMM6000 || COMM1999) && 48UOC"
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:210]: Add in a warning field for good standing

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+    }
+
+def COMM_3030():
+    """
+        "original": "\u2022\tminimum WAM of 65;<br/>\u2022\thave completed a minimum of 48 UOC at the commencement of this course;<br/>\u2022\thave room in their degree for this course;<br/>\u2022\tare in good academic standing;<br/>\u2022\thave completed COMM6000 Career Accelerator: Essentials or COMM1999 First Year Portfolio (Business School students only)<br/>This course may count as a Business School Free Elective, OR as a major elective within some schools. It is recommended a progression check is completed prior to enrolling.<br/><br/>",
+        "processed": "\u2022\t65WAM; \u2022\thave a minimum 48UOC at the commencement of this course; \u2022\thave room in their degree for this course; \u2022\tare in good academic standing; \u2022\thave COMM6000 Career Accelerator: Essentials || COMM1999 First Year Portfolio (Business School only) This course may count as a Business School Free Elective || as a major elective within some schools. It is recommended a progression check is prior to enrolling"
+    """
+
+    return "(COMM6000 || COMM1999) && 65WAM && 72UOC"
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:241]: Should be 48UOC not 72

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+def COMM_3090(conditions):
+    """
+        "original": "Students are expected to be in their final year of a Bachelor of Commerce single or dual degree with at least 108 UOC completed.<br/><br/>",
+        "processed": "are expected to be in their final year of a Bachelor of Commerce single || dual degree with 108UOC"
+    """
+
+    return {
+        "original": conditions["original"],
+        "processed": "108UOC",
+        "warning": "Students are expected to be in their final year of a Bachelor of Commerce single or dual degree with at least 108 UOC completed."
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:252]: Shorten warning to "Students are expected to be in their final year of a Bachelor of Commerce single or dual degree"

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 10]
```diff
@@ -0,0 +1,333 @@
...
+def COMM_3101(conditions):
+    """
+        "original": "Pre-requisite: Completion of 48 UOC.<br/>This course is by application only.Please contact the Co-op office for more information.<br/>Students who have completed DIPP1510 or COMM2222 or are in Business Co-Op programs are excluded from this course.<br/><br/>",
+        "processed": "48UOC. This course is by application only.Please contact the Co-op office for more information. who have DIPP1510 || COMM2222 || are in Business Co-Op programs are"
+    """
+
+    return {
+        "original": conditions["original"],
+        "processed": "48UOC",
+        "warning": "This course is by application only.Please contact the Co-op office for more information. Students who have completed DIPP1510 or COMM2222 or are in Business Co-Op programs are excluded from this course."
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:264]: Take out the DIPP1510 and COMM2222 and add to exclusions

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 11]
```diff
@@ -0,0 +1,333 @@
...
+    }
+
+def COMM_3202(conditions):
+    """
+        "original": "Pre-requisite: COMM3101 or COMM2101<br/>Both terms of COMM3101/COMM2101 (12 UOC) must be successfully completed. <br/>This course is by application only. Please contact the Co-op office for more information.<br/>Excluded:DIPP1510, COMM2222, Business Co-Op programs<br/><br/>",
+        "processed": "COMM3101 || COMM2101 terms of (COMM3101 || COMM2101) ( 12UOC) must be successfully . This course is by application only. Please contact the Co-op office for more information"
+    """
+
+    COURSES["COMM3202"]["exclusions"]["DIPP1510"] = 1
+    COURSES["COMM3202"]["exclusions"]["COMM222"] = 1
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:274]: Missing a 2 here

File: backend/data/processors/manualFixes/COMMFixes.py [hunk 12]
```diff
@@ -0,0 +1,333 @@
...
+def COMM_3999(conditions):
+    """
+        "original": "Pre-requisite: Completed myBCom First Year Portfolio (COMM1999) and in their final year of a single or double Commerce degree (completed at least 72 UOC of Business courses).<br/><br/>",
+        "processed": "myBCom First Year Portfolio (COMM1999) && in their final year of a single || double Commerce degree ( 72UOC of Business courses)"
+    """
+
+    return {
+        "original": conditions["original"],
+        "processed": "COMM1999 && 72UOC && COMM#",
+        "warning": "Studetns must be in their final year"
```
Comment (JamesyJi) [inline backend/data/processors/manualFixes/COMMFixes.py:322]: Misspelt students in warning

Change 72UOC to `72UOC in ZBUS`

**Review by JamesyJi — CHANGES_REQUESTED** [review]
Business school courses r just doomed honestly gg

**Comment by lhjt** [comment]
Closing for renovate.

**Pull request merged**